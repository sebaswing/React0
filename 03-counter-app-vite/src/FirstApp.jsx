// import { Fragment } from "react";
// const GetName = (a,b) => {
//     return a + b;
// }

import PropTypes from 'prop-types';

export const FirstApp = ({ title, subtitle='No hay subtítulo',name='No hay título' }) => {

// if (!title) {
//     throw new Error("El título no existe");
// }

    // console.log(props);
    return (
        // <Fragment>
        //<>   es lo mismo que un Fragment
        <> 
            {/* <h1>{ GetName(2,3) }</h1> */}
            <h1 data-testid="test-title">{title}</h1>
            <p>{subtitle}</p>
            <p>{subtitle}</p>
            <p>{name}</p>
        </>
        // </Fragment>
    );
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string,
}

// FirstApp.defaultProps = { deprectado
//     name: 'Sebastián Beltrán', // Propiedad por defecto,
//     subtitle: 'No hay subtítulo',
//     // title: 'No hay título',
// }
