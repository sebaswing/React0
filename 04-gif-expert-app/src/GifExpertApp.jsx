export const GifExpertApp = () => {
  return (
    <>
    {/*titulo  */}
      <h1>GifExpertApp</h1>

      {/*input */} 


      {/*listado de gifs */} 
       {/*gif item */}

    </>
  )
}
