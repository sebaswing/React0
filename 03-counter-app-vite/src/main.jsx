import React from "react";
import ReactDOM from "react-dom/client";
// import { HelloWorldApp } from "./HelloWorldApp";
import { FirstApp } from "./FirstApp";
import "./styles.css";
import { CounterApp } from "./CounterApp";



ReactDOM.createRoot(document.getElementById("root")).render(
    <React.StrictMode>
        {/* <HelloWorldApp /> */}
        {/* <FirstApp title='Hola, Soy SebastiÃ¡n' /> */}
        <CounterApp value= { 20 } />
    </React.StrictMode>
)