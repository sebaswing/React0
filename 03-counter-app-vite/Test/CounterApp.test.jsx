import { render,screen } from "@testing-library/react";
import { CounterApp } from "../src/CounterApp";

describe('pruebas para CounterApp',()=>{


    //TODO hacer match con el snapshot
    
    test('debe probar que el snapshot sea igual', () => {  
        const { container } = render( <CounterApp value={100} /> );
        expect( container ).toMatchSnapshot();

    });

    //TODO debe mostrar el valor inicial de 100 <CounterApp value={100} />
    test('debe probar que el valor inicial sea 100', () => { 
        render( 
        <CounterApp value={100} /> 

        );
        expect(screen.getByText(100)).toBeTruthy();
        // expect(screen.getByRole('heading',{level:2}).innerHTML).toContain('100');

     });

});