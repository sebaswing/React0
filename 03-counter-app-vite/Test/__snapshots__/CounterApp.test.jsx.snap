// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`pruebas para CounterApp debe probar que el snapshot sea igual 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    100
     
  </h2>
  <button>
    +1
  </button>
  <button>
    -1
  </button>
  <button>
    Reset
  </button>
</div>
`;
